### ISSUE TITLE
Ruler manager doesn't reflect `appender.Commit()` errors in the metrics

### ISSUE THREAD

**Issue by simonswine** [prometheus/prometheus#8386]
## Proposal

**Use case**:

Investigating a Cortex ruler problem recently. I noticed that the ruler manager doesn't reflect errors from `appender.Commit` in the ruler evaluation metrics:
https://github.com/prometheus/prometheus/blob/b7fe028740b7b36a31c2deda1e2b74aa566fc0ee/rules/manager.go#L618

While the errors are handled for `appender.Add` ( and reflected correctly in the metrics there is no accounting for errors that happen on  `appender.Commit`:
https://github.com/prometheus/prometheus/blob/b7fe028740b7b36a31c2deda1e2b74aa566fc0ee/rules/manager.go#L625

The Cortex  `Appendable` isn't able to verify problems with the samples during `Add()`, so out-of-order or duplicate metrics will only fail at `Commit` time.

**Proposal**:

Errors during a `Commit` should be accounted for with a similar logic as for `Add`, the relevant error counter needs to be increased by the value of successful `Add` calls

**Comment by brian-brazil** [comment]
Would you like to send a PR to add metrics for this? The existing metric is correct, as it only talks about evaluation failures - and an ingestion failure is not an evaluation failure.

**Comment by simonswine** [comment]
Yes agreed, I am happy to work on a PR for this. 

Do you think it's worth it having a condition label value like that:

```
rule_sample_ingestion_total[rule_group]
rule_sample_ingestion_failures_total[rule_group,condition='out-of-order|duplicate|other']
```

**Comment by brian-brazil** [comment]
Separate metrics seem best here, to match what we have for scraping.

**Comment by simonswine** [comment]
Closed through #8770
